### PR TITLE
text2image api route

### PR DESCRIPTION
**Pull request by HTSagara**
## How to test it
- Login to you account.
- Copy your `access_token` and paste it to the **Postman**
- Set your **Headers** with `Content-Type` and `application/json`
- **Body** should be a JSON:
```JSON
{
    "prompt": "A futuristic city at sunset"
}
```
Wait until it creates the image. It will display on **Postman**

### CODE REVIEW

**Comment by HTSagara** [comment]
issues #141 amd #142

**Comment by HTSagara** [comment]
## Overview
API updated to use hf spaces and save image on s3.

To test it is the same way as before. But you will need a hf token able to access the  in your .env file.
Make sure to add the Repositories permissions "spaces/htsagara/stabilityai-stable-difusion...." when creating the token.

Screenshot 2024-10-28 at 9 48 36 AM

The API will return a link to download the image from S3.

File: backend/src/routes/text2image.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+                img_data = img_file.read()
+        else:
+            raise HTTPException(status_code=500, detail="Unexpected response format or file not found")
+
+        # Prepare file name and S3 path
+        file_name = f"{prompt.prompt[:10].replace(' ', '_')}_generated_image.png"
+        s3_key = f"generated_images/{file_name}"
+
+        # Upload the image data to S3 with public-read access
+        s3_client.upload_fileobj(io.BytesIO(img_data), S3_BUCKET_NAME, s3_key, ExtraArgs={'ACL': 'public-read'})
```
Comment (JonPeeAir) [inline backend/src/routes/text2image.py:57]: remove `ExtraArgs={'ACL': 'public-read'}` because ACL is disabled

Comment (HTSagara) [reply backend/src/routes/text2image.py]: done S2

**Pull request merged**